qiita_db/portal.py

```diff
@@ -32,32 +33,156 @@ def get_studies_by_portal(portal):
         return {x[0] for x in
                 conn_handler.execute_fetchall(sql, [portal_id])}
 
-    def add_study_to_portal(study, portal):
-        """Adds study to given portal
+    def add_studies_to_portal(portal, studies):
+        """Adds studies to given portal
 
         Parameters
         ----------
+        studies : list of int
+            Study ids to attach to portal
         portal : str
             Portal to associate with.
         """
         portal_id = convert_to_id(portal, 'portal_type', 'portal')
+        conn_handler = SQLConnectionHandler()
+        sql = """SELECT study_id from qiita.study_portal
```

Comment:
The query is doing exactly the same thing as your query above, just giving a "clean" list of studies instead of a "dirty" list of duplicates. Hence why a warning is OK in my mind, since we are only operating on the cleaned list of studies.